dimagi/commcare-hq
# Update web app permissions to app access permissions

https://dimagi-dev.atlassian.net/browse/ICDS-1500
##### SUMMARY

The change will make the access permissions consistent for web apps and mobile apps which will allow us to use it for restricting users on mobile as well. 

We are updating permissions and adding two new permissions `access_all_apps ` and `allowed_apps_list` and a new feature flag `ROLE_APP_ACCESS_PERMISSIONS `. 

We are aiming to remove existing permission `view_web_apps` and `view_web_apps_list` and the feature flag `ROLE_WEBAPPS_PERMISSION`. 

This is the first part of the process where-
1) I have added new permissions and feature flag
2) Updated HTML and JS to use these new permissions
3) Migration to copy existing permissions to the new ones.

I will be following up with another PR #28576  that will focus on removing unused thigs from code.
1) Remove unused fields and from models.
2) Updates the code to use new functions instead of old ones.
3) Migrations to delete removed attributes from DB and migrate the feature flag.
##### FEATURE FLAG

ROLE_APP_ACCESS_PERMISSIONS
##### RISK ASSESSMENT / QA PLAN

Before we deploy it, we would want that-
- We have at least 2 domains in which `ROLE_WEBAPPS_PERMISSIONS` is enabled. 
- We will set up a role that will have access to a specific web app in that domain.
- We will set up a role that will have access to all web apps in that domain.
- We will assign the roles to two different test users.
- We will make sure that users are able to access applications based on the role.

Now we will deploy this PR to staging env.
- Check if we have a new feature flag `ROLE_APP_ACCESS_PERMISSIONS` and we will enable it for the domain that we have used before.
- We will test if the UI is updated and Web apps permissions are changed to app access permissions.
- We will verify that the permissions that we had given for different roles are successfully migrated.
- Verify if we change the permissions they are reflected correctly.
- Verify that users are able to login correctly.

Then we will deploy the followup PR and we should verify all the things that we did in the previous part and also
- Test that the domains that were there in `ROLE_WEBAPPS_PERMISSIONS` are now in `APP_ACCESS_PERMISSIONS`.
##### PRODUCT DESCRIPTION
